### ISSUE TITLE
How should we improve grubs?

### ISSUE THREAD

**Issue by SpadesNeil** [VOREStation/VOREStation#6047]
So a lot of people don't like grubs, but a lot of people love grubs. We refuse to delete them, but there's clearly ways to make them better. This issue post is to discuss those ideas.

#### Ideas so far
- Make grubs only disable a single APC and not drain the whole grid.
- Make grubs provide some kind of better reward. (Still need ideas for this.)

#### Code Revision
- https://github.com/VOREStation/VOREStation/commit/df79f20be7e97617797166c7b7f78b4597812e7f
 
#### Anything else you may wish to add:
Continuation of https://github.com/VOREStation/VOREStation/issues/6043

**Comment by chaoko99** [comment]
Another thing pointed out was to make them raise power alarms in their relevant locations. However, having them straight-up EMP the APC will achieve both things in fewer calls... Though I don't know how long.

**Comment by Wickedtemp** [comment]
A few ideas.

- Limit them to areas with 3+ Active Personnel
- Limit them to rounds with 3+ Engineering Crew
- Put a cap on how far they can drain power
- Having them only target a single APC would be a good idea
- Current 'reward' is 200v, used in Vermicetol, a slightly improved brute-healing chem that isn't sought after. I'm not sure what to add that would make me actually want to deal with grubs other than to stop the nuisance. If it could be converted into slime jelly, carp or spider toxins with various recipes, that might help. There's also a reagent being added on Polaris that they'd acquire from a plant on their map, could have 200v be in a recipe to create that reagent since we wouldn't have the plant here. Could also have grub-meat be valuable for cargopoints, like ores.
- Assistance in locating grubs would be helpful. I don't care if its Eng-only via power alarms or if there was some other way where a department could check and locate any grubs without, for example, having to sweep the several-floor Civ areas and hoping they didn't forget a spot.

**Comment by SpadesNeil** [comment]
Bringing the scope down from crippling an entire grid with a power sink to something like just breaking the power in a single area by way of rapidly draining that lone APC sounds like it would fix many problems at once.

Polaris adding in that new reagent could be good too. Need more lore on what their reagent does. However I'd like to come up with something more original if possible.

**Comment by Wickedtemp** [comment]
TLDR what they're thinking is that its a reagent that could effectively convert carptoxin to spidertoxin and back again, but that really isn't useful because we don't really get either of those reagents anyway. If carptoxin, spidertoxin and slimejelly each had recipes and could be made from this reagent, then that'd actually be both useful and still limited and not an every-round thing. The reagent is Sivian Sap, not sure if its actually been added to Polaris code yet.
https://imgur.com/S7ee1aM

**Comment by Ketrai** [comment]
Well I'll refer back to my earlier suggestions. 

A. Introduce an inert grub larvae event. Grubs have entered the station but their malicious ai is disabled and the larva never grow up as a result. Lore wise they're either genetic rejects or just they're just so young they're not ready to drain energy yet. (White hue). Make the actual grub event that's annoying disabled by natural means. Only having a 10% chance to be triggered when the inert grub larvae event happens. Aka a herald to the rest of a hive party arriving. The main complaint I've seen so far is them being too frequent. (50%~) of rounds. And the event manager being too thin.

B. Change them from apc drain to causing ghost lights in an area or small malfunctions to equipment when being used. This requires more coding making it more annoying, but we already have area-specific light flicker effects in the code for lamp breaking events. For equipment you can just make it target the user giving them a 25% chance for a harmless stun or something. Annoying enough to have it fixed. The main difference with normal grubs here being that they are more contained. And only affect the area they're in directly. Could make them a variant so the old original grubs can still be adminspawned. 

C. Add a science bound device that can be printed to easily track down grubs. So engineers aren't expected to be the poor sod that has to help out. Science can just lay one of these on their desk at the start of the shift. (It can spawn in the reception.) Could be specified as detecting power surges but being refitted specifically for modern virgo pests.

**Comment by Ketrai** [comment]
As for better rewards? My dude just make it so you can ship them on the cargo shuttle. Two lines of code and immediately any cargo player will dedicate them to the cause xD. Even better. Make it so you get ore as well as meat upon gibbing/grinding. So miners get points whilst cargo also gets points. If you give me some time I'd happily implement this.

**Comment by Ketrai** [comment]
As for other suggestions. Most of what wicked suggests is quite hard to code except the power limit. Already exists in larvae code kinda. Just make grubs evolve further into a fatty satisfied version, that just roots itself in its corner to transfer power back to their queen. Easy to code, lowers the time period of disruption to the point you can ignore them. 

Periodic emps are also easy to code. But that doesn't take away the apparent obligation for engineers to take care of it. Just makes them easier to find.

**Comment by Very-Soft** [comment]
I think making them drain just the apc they're on rather than the grid might make them more annoying to deal with actually, since, if they're draining from the grid, eventually you'll have all the power in the area go out, and anyone with a crowbar can just run in and fight them. 

If they only affect the room they're in and no one's around to let you in, then it makes breaking in via hacking or deconstructing stuff basically required to deal with them.

Not that that's a huge problem, but it sounds more annoying to deal with to me.

**Comment by Ketrai** [comment]
I also like the idea of them dropping a sparkly donut called a sparklenut. Flavoured as tasty remnants of digestive function. These can have 5u of vermicetol and little nutrition to make the chemical easier to obtain and immediately useful. Can make it an expedition food that ignores helmets too.

**Comment by Very-Soft** [comment]
Also I want to say.

I think it'd be neat to have several variants of the Grub event with essentually similar spawn parameters but that spawn different mobs. 

This could be gaslamps or geese or mimics or any number of other things hostile or not, as long as individually they're not too extremely threatening. And make it so that only -one- of that kind of event can happen in a shift. That way if a few geese get in, you know for a fact that no grubs will spawn that shift.

**Comment by Very-Soft** [comment]
I also think it might improve grubs ignorability if they had some kind of sated state. Like. They'll drain an area for a while, but if there's really no one around willing to deal with them, perhaps they get full and stop it eventually. (and harder to kill?)

**Comment by TheDavestDave** [comment]
But if they just drain one APC they are easier to find in the first place, and can be ignored if somewhere not important and engineers don't want to work. Or there are no active engineering

**Comment by Ketrai** [comment]
Dave. When it comes to notification just periodically causing a mini emp should cause an alert on the specific apc nearby. Or other electrical devices.

**Comment by SpadesNeil** [comment]
>But if they just drain one APC they are easier to find in the first place, and can be ignored if somewhere not important and engineers don't want to work. Or there are no active engineering

This is my logic on the matter.

>Dave. When it comes to notification just periodically causing a mini emp should cause an alert on the specific apc nearby. Or other electrical devices.

I hesitate to give them an EMP ability because it fucks up equipment and synth players especially. Got a mechanical heart? RIP you.

**Comment by Wickedtemp** [comment]
...don't they already have an emp ability.

**Comment by SpadesNeil** [comment]
oh, well, I haven't gotten any complaints so idk

I think the EMP is limited to attacks since synths are tanky. Not, EMP randomly every few minutes.

**Comment by Ketrai** [comment]
Making them drain one apc is a little more difficult. They'd have to target the specific area and behave basically like a machine does. I suppose it isn't that much of a code change though, if it's as easy as targeting the power grid. But I'm just scared this would make grubs so neglible of a hassle. You'd only really care if they somehow got in the engine room or one of the 10 rooms actually used on station. The rest is just maintenance and other garbage.

now if they periodically drain the closest apc, moving on once an apc reaches a threshold of 10%, just to drain another one at 50% percent, and they do it fairly quick? Now that would still make em hunt worthy.

**Comment by chaoko99** [comment]
This isn't how development works, stale bot.
